src/axom/core/Macros.hpp

```diff
@@ -35,6 +35,29 @@
 #define AXOM_HOST
 #endif
 
+/*
+ * \def AXOM_PRAGMA
+ *
+ * \brief Macro used to specify pragma directive
+ */
+#ifdef _WIN32
+#define AXOM_PRAGMA(__supplied_pragma) __pragma(__supplied_pragma)
+#else
+#define AXOM_PRAGMA(__supplied_pragma) _Pragma(#__supplied_pragma)
```

Comment:
Does the `#` work here correctly to handle the quotes? I recall I had issues in the past with something similar...

Reply:
Here is what RAJA does:
```
#define RAJA_STRINGIFY(x) RAJA_DO_STRINGIFY(x)
#define RAJA_DO_STRINGIFY(x) #x
#ifdef _WIN32
#define RAJA_PRAGMA(x) __pragma(x)
#else
#define RAJA_PRAGMA(x) _Pragma(RAJA_STRINGIFY(x))
#endif
```

The `STRINGIFY` macro is weird, but, it handles the quotes correctly....

We should follow that same pattern here.

Reply:
Added!